### PR TITLE
Further gammapy.image cleanup

### PR DESCRIPTION
**Pull request by adonath**
This PR contains the following changes:
* Remove `BoundingBox` class and docs 			
* Raise `WcsError` in SkyMap.paste(), when WCSs not aligned 
* Add `.contains()` method to `SkyMap` and remove from utils

### CODE REVIEW

**Comment by cdeil** [comment]
There's one fail on travis-ci (that you also should see locally after running `make clean`):
https://travis-ci.org/gammapy/gammapy/jobs/136725513#L941
Please fix.

File: gammapy/image/maps.py [hunk 1]
```diff
@@ -313,12 +331,16 @@ def paste(self, skymap, method='sum'):
...
-        else:
-            raise ValueError('Invalid method: {}'.format(method))
+        try:
+            if method == 'sum':
+                self.data[ylo:yhi, xlo:xhi] += skymap.data[ylo_c:yhi_c, xlo_c:xhi_c]
+            elif method == 'replace':
+                self.data[ylo:yhi, xlo:xhi] = skymap.data[ylo_c:yhi_c, xlo_c:xhi_c]
+            else:
+                raise ValueError('Invalid method: {}'.format(method))
+        except ValueError:
```
Comment (cdeil) [inline gammapy/image/maps.py]: This try-except logic doesn't look right?
If the user passes `method='spam'` you raise a ValueError, then catch it, and raise a WCSError?

**Comment by cdeil** [comment]
I left two inline comments.

👍 , although I'm sad to see the `BoundingBox` class go. :-)

File: gammapy/image/maps.py (hunk 1 above)
Comment (adonath) [inline gammapy/image/maps.py]: Indeed that was nonsense. Fixed

**Comment by cdeil** [comment]
Are you sure that `contains` still works for all common use cases?
I.g. scalar, 1d and 2d arrays of test positions?
Positions inside and outside the image?
And test and image in different systems?

For me it's fine to merge as-is if you think it's OK.
Maybe leave a TODO comment for Olga that tests should be expanded for this?

**Comment by cdeil** [comment]
Looks good ... merging this now.
Thank you!

**Pull request merged**